transform_hoist_assign.ml

```diff
@@ -13,12 +13,13 @@ open Instr
  * We only look at our own use-def chain. Thus the transformation renames the
  * variable to avoid overriding unrelated uses of the same name.
  *)
-let hoist_assignment (instrs : instruction_stream) : instruction_stream option =
-  let reaching = Analysis.reaching instrs in
-  let uses = Analysis.used instrs in
-  let dominates = Analysis.dominates instrs in
+let hoist_assignment ((formals, instrs) as inp : analysis_input) : analysis_input option =
+  let open Analysis in
+  let reaching = reaching inp in
+  let uses = used inp in
```

Comment:
`uses` would be consistent with `dominates` (and also the comment). But then we also have `required`, `required_vars`, and `aliased`. (I'm referring to the definitions in `analysis.ml`.)

At the very least, in this part of the code `uses` is consistent with how the variable is used (e.g. `dominates_all_uses`).